portal/api-user-grpc.go

```diff
@@ -51,3 +52,36 @@ func (s *server) AddUser(ctx context.Context, in *pb.AddUserRequest) (*pb.User,
 
 	return &pb.User{Name: newUser.Name, Email: newUser.Email}, nil
 }
+
+func (s *server) ListUsers(ctx context.Context, in *pb.ListUsersRequest) (*pb.ListUsersResponse, error) {
+	// Validate sessionID and get tenant short name using the valid sessionID
+	tenantShortName, err := getTenantShortNameFromSessionID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	reqOffset := in.GetOffset()
+	reqLimit := in.GetLimit()
+	if reqLimit == 0 {
+		reqLimit = defaultRequestLimit
+	}
+	appCtx, err := cluster.NewContext(tenantShortName)
+	if err != nil {
+		return nil, err
+	}
+	// Get list of users set maximum 25 per page
+	users, err := cluster.GetUsersForTenant(appCtx, reqOffset, reqLimit)
+	if err != nil {
+		return nil, status.New(codes.Internal, "Error getting Users").Err()
+	}
+	// Get total of users
+	total, err := cluster.GetTotalNumberOfUsers(appCtx)
```

Comment:
We could use a different pagination technique later where we can avoid getting the total number of users. For now this works.